torch/testing/_internal/common_methods_invocations.py

```diff
@@ -672,6 +672,7 @@ def method_tests():
         ('index_add', (S, S), (0, index_variable(2, S), (2, S)), 'dim', (), [0]),
         ('index_add', (), (0, torch.tensor([0], dtype=torch.int64), (1,)), 'scalar_input_dim', (), [0]),
         ('index_add', (), (0, torch.tensor(0, dtype=torch.int64), ()), 'scalar_all_dim', (), [0]),
+        ('index_add', (S, S), (0, index_variable(2, S), (2, S)), 'dim_alert_nondeterministic', (), [0], [expectedAlertNondeterministic('cuda', 'index_add_cuda_')]),
```

Comment:
I figured out how to make a decorator that turns on determinism and checks that the correct error is thrown. But now I notice that the pooling and sampling operations that use `atomicAdd` don't seem to be a part of this `method_tests()` list. Would it be alright for me to add them here, or might they be somewhere else? I tried grepping for any other places in the tests that use a similar mechanism and include the pooling and sampling operations, but I didn't find them.

Reply:
Traditionally the nn modules/functions have been tested slightly differently, see common_nn.py.